nifi-nar-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/poi/ConvertExcelToCSVProcessor.java

```diff
@@ -199,93 +166,52 @@ public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
         final FlowFile flowFile = session.get();
-        if ( flowFile == null ) {
+        if (flowFile == null) {
             return;
         }
 
-        final String desiredSheetsDelimited = context.getProperty(DESIRED_SHEETS).evaluateAttributeExpressions(flowFile).getValue();
+        final Map<String, Boolean> desiredSheets = getDesiredSheets(context, flowFile);
```

Comment:
Wondering if we could use the primitive `boolean` type here, based on a cursory read-through it doesn't appear we need the Object version here

Reply:
No a primitive cannot be used here. The Map represented with desiredSheets keeps track of which specified sheets were found (lines 189-196)